pkg/lexer/lexer.go

```diff
@@ -9,10 +9,19 @@ import (
 	"github.com/bragov4ik/yacal/pkg/lexer/tok"
 )
 
+func isAtomEnd(r rune) bool {
+	return unicode.IsSpace(r) || r == lex.EOF || r == ')'
```

Comment:
```suggestion
	return unicode.IsSpace(r) || r == lex.EOF || r == ')' || r == '('
```
Or maybe even
```suggestion
	return !unicode.IsLetter(r)
```
Case `(setq x(+ 2 3))` parses `x(+` as literal

Reply:
![image](https://user-images.githubusercontent.com/8144358/151239379-35e376ec-7441-44d5-a6e6-934abc6af7a4.png)
From grammar definition (in project description)